client-spark/spark3/src/main/java/org/apache/spark/shuffle/writer/RssShuffleWriter.java

```diff
@@ -135,22 +150,60 @@ private boolean isMemoryShuffleEnabled(String storageType) {
   public void write(Iterator<Product2<K, V>> records) throws IOException {
     List<ShuffleBlockInfo> shuffleBlockInfos = null;
     Set<Long> blockIds = Sets.newConcurrentHashSet();
+
+    Function1<Tuple2<K, V>, Void> spillFunc = new AbstractFunction1<Tuple2<K, V>, Void>() {
+      @Override
+      public Void apply(Tuple2<K, V> tuple) {
+        int partition = getPartition(tuple._1);
+        List<ShuffleBlockInfo> shuffleBlockInfos = bufferManager.addRecord(partition, tuple._1, tuple._2);
```

Comment:
Method `addRecord` also need memory. If we can release a record memory after we call method `addRecord`. It will be better.

Reply:
Should we put map side combine operation into WriteBufferManager? When we need to spill, we can send the record directly?